test/utils.py

```diff
@@ -665,6 +665,13 @@ def joinall(threads):
         assert not t.is_alive(), "Thread %s hung" % t
 
 
+async def asyncjoinall(tasks):
+    """Join tasks with a 5-minute timeout, assert joins succeeded"""
+    for t in tasks:
+        await asyncio.wait([t.task], timeout=300)
```

Comment:
Let's use gather() to join all tasks at once.

Reply:
Better yet, use `asyncio.wait`.

Reply:
good idea! done!